### ISSUE TITLE
Unable to start server in batch only mode

### ISSUE THREAD

**Issue by mattkallo** [spring-cloud/spring-cloud-dataflow#4948]
**Description:**
Starting server in batch only mode terminates with CloudFoundry configuration issue. As per my understanding, CloudFoundry is not required in batch only mode. Please correct me if I am wrong.

`Binding to target org.springframework.boot.context.properties.bind.BindException: Failed to bind properties under 'spring.cloud.deployer.cloudfoundry' to org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryConnectionProperties failed:
`

**Release versions:**
spring-cloud-dataflow-server:2.9.4

**Custom apps:**
None.

**Steps to reproduce:**
_Followed the steps in_ 
https://dataflow.spring.io/docs/recipes/batch/batch-only-mode/
- Properties were set as below
- Started only Spring Cloud Data Flow Server

With properties set as 
```
spring.cloud.dataflow.features.streams-enabled=false
spring.cloud.dataflow.features.schedules-enabled=false
spring.cloud.dataflow.features.tasks-enabled=true
```

**Additional context:**
Log

```
 o.f.c.internal.license.VersionPrinter    : See what's new here: https://flywaydb.org/documentation/learnmore/releaseNotes#8.5.11
 o.f.c.internal.license.VersionPrinter    :
 o.f.c.i.database.base.BaseDatabaseType   : Database: jdbc:mysql://10.128.0.34:3306/dataflow (MySQL 8.0)
 o.f.core.internal.command.DbValidate     : Successfully validated 4 migrations (execution time 00:00.578s)
 o.f.core.internal.command.DbMigrate      : Current version of schema `dataflow `: 4
 o.f.core.internal.command.DbMigrate      : Schema `dataflow` is up to date. No migration necessary.

 org.hibernate.Version                    : HHH000412: Hibernate ORM core version 5.6.9.Final
 o.hibernate.annotations.common.Version   : HCANN000001: Hibernate Commons Annotations {5.1.2.Final}
 org.hibernate.dialect.Dialect            : HHH000400: Using dialect: org.hibernate.dialect.MySQL8Dialect

 j.LocalContainerEntityManagerFactoryBean : Initialized JPA EntityManagerFactory for persistence unit 'default'
: Unsatisfied dependency expressed through method 'connectionContext' parameter 0; nested exception is org.springframework.boot.context.properties.ConfigurationPropertiesBindException: Error creating bean with name 'cloudFoundryConnectionProperties': Could not bind properties to 'CloudFoundryConnectionProperties' : prefix=spring.cloud.deployer.cloudfoundry, ignoreInvalidFields=false, ignoreUnknownFields=true; nested exception is org.springframework.boot.context.properties.bind.BindException: Failed to bind properties under 'spring.cloud.deployer.cloudfoundry' to org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryConnectionProperties
 j.LocalContainerEntityManagerFactoryBean : Closing JPA EntityManagerFactory for persistence unit 'default'
 com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown initiated...
 com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown completed.


Description:

Binding to target org.springframework.boot.context.properties.bind.BindException: Failed to bind properties under 'spring.cloud.deployer.cloudfoundry' to org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryConnectionProperties failed:

    Property: spring.cloud.deployer.cloudfoundry.org
    Value: null
    Reason: must not be null

```

**Comment by mheath** [comment]
This is fixed in both 2.10.0-M2 and 2.9.6.

**Comment by xiamubobby** [comment]
I think it's occurring again in 2.10.3...
same config as the original ticket, same issue
